util/reggen/field.py

```diff
@@ -328,3 +377,30 @@ def sw_readable(self) -> bool:
 
     def sw_writable(self) -> bool:
         return self.swaccess.key != 'ro'
+
+    def apply_alias(self, alias_field: object, where: str) -> None:
+        '''
+        Compare all attributes and replace overridable values.
+
+        This updates the overridable field attributes with the alias values and
+        ensures that all non-overridable attributes have identical values.
+        '''
+        if not isinstance(alias_field, Field):
+            raise ValueError('alias_field is not of type Field')
+
+        for key, val in self.__dict__.items():
+            # This is an overridable attribute, so replace the value.
+            if key in ALIAS_FIELD_ATTRS:
+                setattr(self, key, getattr(alias_field, key))
+            # This is not an overridable attribute, so ensure values match.
+            elif key in NON_ALIAS_FIELD_ATTRS:
```

Comment:
This `elif` is what I was complaining about in the definition of the two lists earlier.

Also, I'm not a huge fan of `getattr` and `setattr`: they defeat static typing and are generally a pain when grepping around. I think it would probably be better just to write out the list of fields explicitly (you could put the comparison check in a helper function to avoid having to repeat the `ValueError` stuff)

Reply:
Jep, as mentioned, I've reworked this a bit - see fixup.